### DISCUSSION TITLE
How to set up a non-uniform boundary condition?

### DISCUSSION THREAD

**Discussion by waaizhaq** [idaholab/moose#29552] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello,

I have a rectangular mesh where I set the bottom boundary to a NeumannBC of value 0, representing the ground, the right and top walls to a DirichletBC of value 296.15 Kelvin, representing the ambient air, and currently I have the left wall set to a DirichletBC of value 505.15 Kelvin. However, I want to change that to where only the lower half of the wall is set to the 505.15 Kelvin, and the upper half is another insulated boundary, or NeumannBC of value 0.

**Comment by snowbugxs** [comment]
I think you may need a section on the domain, then you get two left walls, lower one and upper one. Set different BCs to them.

**Reply by waaizhaq** [reply]
Do you know an example that I can look at? I was trying to find something on the MOOSE framework but couldn't find what I was looking for.

**Reply by GiudGiud** [reply]
Hello
Use this generator to split the sideset in two. Then use one boundary condition on one part and another on the other part
https://mooseframework.inl.gov/source/meshgenerators/ParsedGenerateSideset.html

**Reply by snowbugxs** [reply]
Can we do it outside moose, e.g, using Gmsh to split the domains?

**Reply by GiudGiud** [reply]
Yes. You can take any mesh generation step in meshing software outside of moose
